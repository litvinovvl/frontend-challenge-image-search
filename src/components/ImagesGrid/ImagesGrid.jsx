import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { ReactComponent as NoResultsIcon } from '../../assets/images/sad-face.svg';
import { ReactComponent as LoadingIcon } from '../../assets/images/think-face.svg';

import './style.scss';

const ImagesGrid = ({ images, isLoading, error }) => {
  const renderImages = () => {
    if (error) {
      return (
        <div className="spinner-wrapper">
          <NoResultsIcon width={48} hanging={48} />
          Something went wrong, try again
        </div>
      );
    }

    if (isLoading) {
      return (
        <div className="spinner-wrapper">
          <LoadingIcon width={48} hanging={48} />
          Loading
        </div>
      );
    }

    if (images && images.length) {
      return (
        <div className="images-wrapper">
          {
            images.map((image) => (
              <img key={image.id} src={image.webformatURL} alt={image.tags} />
            ))
          }
        </div>
      );
    }

    return (
      <div className="spinner-wrapper">
        <NoResultsIcon width={48} hanging={48} />
        No results found!
      </div>
    );
  };

  return (
    <div className={classNames('container', { hidden: !images })}>
      {renderImages()}
    </div>
  );
};

ImagesGrid.defaultProps = {
  images: null,
  isLoading: false,
  error: false,
};

ImagesGrid.propTypes = {
  images: PropTypes.arrayOf(PropTypes.object),
  isLoading: PropTypes.bool,
  error: PropTypes.bool,
};

export default ImagesGrid;
