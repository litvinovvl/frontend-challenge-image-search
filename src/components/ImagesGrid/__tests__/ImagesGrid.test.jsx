import React from 'react';
import { shallow } from 'enzyme';

import ImagesGrid from '../ImagesGrid';

const image = { webformatURL: 'test_url', tags: 'tags' };
const images = [
  { ...image, id: 1 },
  { ...image, id: 2 },
];

const props = {
  isLoadding: false,
  error: false,
  images,
};

describe('App', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render ImagesGrid properly initially', () => {
    const component = shallow(<ImagesGrid {...props} images={null} />);

    expect(component).toMatchSnapshot();
  });

  it('should render ImagesGrid properly when loading', () => {
    const component = shallow(<ImagesGrid {...props} isLoading />);

    expect(component).toMatchSnapshot();
  });

  it('should render ImagesGrid properly when error', () => {
    const component = shallow(<ImagesGrid {...props} error />);

    expect(component).toMatchSnapshot();
  });

  it('should render ImagesGrid properly when no images found', () => {
    const component = shallow(<ImagesGrid {...props} images={[]} />);

    expect(component).toMatchSnapshot();
  });

  it('should render ImagesGrid properly when images loaded', () => {
    const component = shallow(<ImagesGrid {...props} />);

    expect(component).toMatchSnapshot();
  });
});
