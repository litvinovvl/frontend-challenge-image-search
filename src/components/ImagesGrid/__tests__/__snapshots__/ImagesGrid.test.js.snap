// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render ImagesGrid properly initially 1`] = `
<div
  className="container hidden"
>
  <div
    className="spinner-wrapper"
  >
    <ForwardRef(SvgSadFace)
      hanging={48}
      width={48}
    />
    No results found!
  </div>
</div>
`;

exports[`App should render ImagesGrid properly when error 1`] = `
<div
  className="container"
>
  <div
    className="spinner-wrapper"
  >
    <ForwardRef(SvgSadFace)
      hanging={48}
      width={48}
    />
    Something went wrong, try again
  </div>
</div>
`;

exports[`App should render ImagesGrid properly when images loaded 1`] = `
<div
  className="container"
>
  <div
    className="images-wrapper"
  >
    <img
      alt="tags"
      key="1"
      src="test_url"
    />
    <img
      alt="tags"
      key="2"
      src="test_url"
    />
  </div>
</div>
`;

exports[`App should render ImagesGrid properly when loading 1`] = `
<div
  className="container"
>
  <div
    className="spinner-wrapper"
  >
    <ForwardRef(SvgThinkFace)
      hanging={48}
      width={48}
    />
    Loading
  </div>
</div>
`;

exports[`App should render ImagesGrid properly when no images found 1`] = `
<div
  className="container"
>
  <div
    className="spinner-wrapper"
  >
    <ForwardRef(SvgSadFace)
      hanging={48}
      width={48}
    />
    No results found!
  </div>
</div>
`;
